src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -41,58 +58,189 @@ public EmailFormatter(String emailMgs, AttachmentHandler attachmentHandler) thro
+        String rfcFormattedText = (String) emailMsgPlusStatus.get(0);
+        JSONObject status = (JSONObject) emailMsgPlusStatus.get(1);
+        status.put(Utils.JSON_ATTR_ITEM_ID, getItemId());
+
+        // Something bad happened in formatting the email
+        if (rfcFormattedText == null) {
+            log.error("Email formatting is not successful");
+
+            status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_ERROR);
+            status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+            emailMsgPlusStatus.set(1, status);
+            return emailMsgPlusStatus;
+        }
+        // Large email size check, if grater than proposed limit attachment will be dropped
+        if (checkMsgSizeMoreThanExpectedLimit(rfcFormattedText)) {
+            log.warn("Attachments are dropped for the message");
+
+            rfcFormattedText = removeAttachments(rfcFormattedText);
+            emailMsgPlusStatus.set(0, rfcFormattedText);
+            status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_PARTIAL);
+            status.put(Utils.JSON_ATTR_MESSAGE, "email Message size exceeds the expected limit, attachments dropped");
+            status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+            emailMsgPlusStatus.set(1, status);
+
+            return emailMsgPlusStatus;
+        }
+
+        JSONArray attachments = status.getJSONArray("attachments");
+        for (int i = 0; i < attachments.length(); i++) {
+            JSONObject attach = attachments.getJSONObject(i);
+            if(attach.get(Utils.MIGRATION_STATUS)!= Utils.STATUS_OK){
+                status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_PARTIAL);
+                status.put(Utils.JSON_ATTR_MESSAGE,"attachments URL from Ctools couldn't be resolved, some or all of the attachments dropped");
+                status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+                emailMsgPlusStatus.set(1, status);
+                return emailMsgPlusStatus;
+            }
+        }
+        // everything went fine at this point
+        status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_OK);
+        status.put(Utils.JSON_ATTR_MESSAGE, Utils.SUCCESS_MSG);
+        status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+        emailMsgPlusStatus.set(1, status);
+        return emailMsgPlusStatus;
     }
 
-    public String rfc822Format() {
+    public List<Object> rfcFormatWithoutAttachmentLimitCheck() {
         ArrayList<String> headers = prunedHeadersWithoutContentType();
-        return getFormattedEmailText(headers).toString();
+        List<Object> formattedEmailText = getFormattedEmailText(headers);
+        return formattedEmailText;
+    }
+
+    public String removeAttachments(String rfcFormattedText) {
+        String emailWithOutattachments = subStringToRemoveAttachments(rfcFormattedText);
+        return emailWithOutattachments;
+    }
+
+    private String subStringToRemoveAttachments(String rfcFormattedText) {
```

Comment:
The name should be more descriptive.  Please give a sentence or two about what is happening in this method unless a better name is sufficient.